rfc-028-adopt-a-code-of-conduct.md

```diff
@@ -0,0 +1,138 @@
+  > complaint relates to the conduct of the Chief Technology Officer, then
+  > complaints may be reported to the Managing Director at
+  > code-of-conduct-md@dxw.com. The current holders of these roles can be found
+  > on [dxw’s people page](https://www.dxw.com/our-people/).
+
+When starting a client project, we SHOULD speak to the client to find out
+whether they would like us to add a code of conduct to the project. We’ll need
+to understand whether they:
+
+- have the ability to accept reports of violations
+- can do this privately
+- want to take on the responsibility of enforcing the code of conduct
+
+dxw SHOULD NOT take on the responsibility of enforcing the Code of Conduct for
+a client project.
+
+TODO Something about enforcement? – this is one to check with Dom and Dave too.
+This is from the Contributor Covenant site:
+
+> Do not simply add the Contributor Covenant to your project and assume that
+> any problems with civility, harassment, or discrimination will be solved.
+> As a project maintainer you must be committed to enforcing the code of
+> conduct. A code of conduct without enforcement sends a false signal that
+> your project is welcoming and inclusive, and can create a dangerous
+> situation for marginalized people who participate. Adding the Contributor
+> Covenant to a project places responsibility on the project team that must
+> not be taken lightly.
+>
+> Before adopting the Contributor Covenant take the time to discuss and
+> decide how to deal with problems as they emerge. Document the policy and
+> procedure for enforcement, and add it to your README or in another visible,
+> appropriate place. Consider if your project team has the willingness and
+> maturity to follow through on your enforcement procedures.
+
+## Concerns
+
+Adding a `CONTRIBUTING.md` might suggest that we’re actively seeking
+contributions from outside dxw, even though sometimes we aren’t. Sometimes we
+create software which we’re happy to share with the world, but which we don’t
+have the time to review contributions for or which we don’t want to grow beyond
+a specific use case. And on client projects, we aren’t paid to review code
+which isn’t to the client’s needs.
+
+This should not prevent us from having a code of conduct – it’s a pretty
+uncontroversial baseline of decent conduct and doesn’t guarantee that we’ll
+review or accept people’s work.  As long as our `CONTRIBUTING.md` is clear
+about whether the project is seeking contributions, and whether we’re likely to
+accept them, then we’re fine. We might like to reply to people’s contributions
+and explain why we’re not going to accept them, even if it takes up a little of
+our time.
+
+## Next steps
+
+- Set up an email contact code-of-conduct@dxw.com that goes to Dominic Baggott
+- Set up an email contact code-of-conduct-md@dxw.com that goes to David Mann
+- Confirm that Dave and Dom are happy to take this role
```

Comment:
@evilstreak (and does Dave have a GitHub account) – I'd like to hear your thoughts on being the reporting / enforcement folks, when you've got a moment to think about this.

Reply:
I've chatted this through with Dave. We're both happy to take on these roles.